### PR TITLE
Make a method that invalidate and updates the intensive quantities

### PR DESCRIPTION
**Pull request by totto82**

### CODE REVIEW

File: opm/models/discretization/common/fvbasediscretization.hh [hunk 1]
```diff
@@ -765,6 +765,29 @@ public:
...
+    {
+        invalidateIntensiveQuantitiesCache(timeIdx);
+
+        // loop over all elements...
+        ThreadedEntityIterator<GridView, /*codim=*/0> threadedElemIt(gridView_);
+#ifdef _OPENMP
+#pragma omp parallel
+#endif
+        {
+            unsigned threadId = ThreadManager::threadId();
```
Comment (atgeirr) [inline opm/models/discretization/common/fvbasediscretization.hh:778]: This variable is not used. Not sure if that is a sign of a bug, or that it should just be removed?

File: opm/models/discretization/common/fvbasediscretization.hh [hunk 2]
```diff
@@ -765,6 +765,29 @@ public:
...
+            ElementContext elemCtx(simulator_);
+            ElementIterator elemIt = threadedElemIt.beginParallel();
+            for (; !threadedElemIt.isFinished(elemIt); elemIt = threadedElemIt.increment()) {
+                // create an element context (the solution-based quantities are not
```
Comment (atgeirr) [inline opm/models/discretization/common/fvbasediscretization.hh:782]: Could you elaborate (preferably in the comment) a little on why the solution-based quantities are not available?

**Review by atgeirr — COMMENTED** [review]
I support the idea here wholeheartedly. I do have a few questions though.

File: opm/models/discretization/common/fvbasediscretization.hh (hunk 2 above)
Comment (totto82) [reply opm/models/discretization/common/fvbasediscretization.hh:782]: copy/paste bug

File: opm/models/discretization/common/fvbasediscretization.hh (hunk 1 above)
Comment (totto82) [reply opm/models/discretization/common/fvbasediscretization.hh:778]: copy/paste bug

**Pull request merged**